### PR TITLE
test: misc test improvements

### PR DESCRIPTION
**Pull request by OrKoN**

### CODE REVIEW

File: test/src/ariaqueryhandler.spec.ts [hunk 1]
```diff
@@ -553,9 +565,14 @@ describe('AriaQueryHandler', () => {
...
-      const waitForSelector = page.waitForSelector('aria/zombo').then(() => {
-        return (divFound = true);
-      });
+      const waitForSelector = page
+        .waitForSelector('aria/zombo')
+        .then(() => {
+          return (divFound = true);
+        })
+        .catch(err => {
+          return err;
```
Comment (Lightning00Blade) [inline test/src/ariaqueryhandler.spec.ts]: On theory error can have a value `true` so return `false` is the correct thing to do here.

Comment (Lightning00Blade) [reply test/src/ariaqueryhandler.spec.ts]: Or add the same expect as the previous test `expect(divHidden).toBe(true);`

File: test/src/elementhandle.spec.ts [hunk 2]
```diff
@@ -389,23 +389,29 @@ describe('ElementHandle specs', function () {
...
-        ElementHandle<HTMLDivElement>
-      >;
+      const innerWaitFor = element.waitForSelector('.bar').catch(err => {
+        return err;
+      }) as Promise<ElementHandle<HTMLDivElement>>;
       await element.evaluate(el => {
         el.innerHTML = '<div class="bar">bar1</div>';
       });
       element = (await innerWaitFor)!;
+      if (element instanceof Error) {
```
Comment (Lightning00Blade) [inline test/src/elementhandle.spec.ts:412]: How about we use `expect(element).not.toBeInstanceOf(Error)` or `expect(element).toBeInstanceOf(ElementHandle)`

Comment (OrKoN) [reply test/src/elementhandle.spec.ts:412]: the stacktrace and the error message is not too detailed in that case imo:

```
  1) ElementHandle specs
       Element.waitForSelector
         should wait correctly with waitForSelector on an element:
     Error: expect(received).not.toBeInstanceOf(expected)

Expected constructor: not Error

      at Context.<anonymous> (/Users/alexrudenko/src/puppeteer/test/src/elementhandle.spec.ts:400:27)
      at processTicksAndRejections (node:internal/process/task_queues:95:5)

```

**Review by Lightning00Blade — APPROVED** [review]

**Pull request merged**